### DISCUSSION TITLE
Please add KoboldCPP support

### DISCUSSION THREAD

**Discussion by FelipeLujan** [open-webui/open-webui#2700] [General]
**Is your feature request related to a problem? Please describe.**
Yes, Some last gen AMD graphics cards are not supported by vanilla Ollama, other implementation have added support for that hardware, such as https://github.com/YellowRoseCx/koboldcpp-rocm
In my case Radeon 6750XT with 12gb vram

**Describe the solution you'd like**
Be able to connect Open webui to a running OpenAI endpoint served by KoboldCPP

**Describe alternatives you've considered**
Forcing Ollama to run on AMD unsupported hardware, without success

**Comment by justinh-rahb** [comment]
Not an issue-worthy topic. You already can, just add it as an "OpenAI" API in **Settings > Connections**.

**Reply by tukangcode** [reply]
Oh so it support koboldcpp? What about horde?

**Reply by Tatalebuj** [reply]
I'm hoping someone can help me get this to work.  I've got "http://localhost:5001" in the OpenAI API box, and nothing in the API Key box, and the subtext says "WebUI will make request to http://localhost:5001/models", yet when I start a new chat I get an error because I have no model loaded.  Which makes sense, as I don't have any models loaded into WebUI, because I want to use my KoboldCPP_ROCM.  But when I type anything in the chat I get an error that no model is loaded.  What am I missing?  How am i supposed to tell Open WebUI to just use the API and ignore the lack of model? Thanks!

**Wanted to add that I'm using a file called "docker-compose.yaml" in the OpenWebui directory and running the command "docker-compose up -d"  

My configuration is:

version: '3'
services:
  open-webui:
    image: ghcr.io/open-webui/open-webui:main
    restart: always
    ports:
      - 3000:8080
    volumes:
      - open-webui:/app/backend/data
    environment:
      - OPENAI_API_BASE_URLS=http://localhost:5001

volumes:
  open-webui:

And the docker is running and Open WebUI opens up on "http://localhost:3000/modelfiles/"

**Reply by justinh-rahb** [reply]
@Tatalebuj you need to set _something_ as the API key, with that blank WebUI will skip that endpoint.

**Reply by Tatalebuj** [reply]
Does KoboldCPP_ROCM have an API Key?  If so, where do I put it so I can then match it in WebUI?  I've just looked through each setting tab in 1.67 Kobold_ROCM and I'm not seeing API key anywhere.  Or do you mean I can put anything I want in there, and the system won't care cause there is no API Key in KoboldCPP?  Thanks for any assistance you can offer!  Cheers,

Update:  Just got kobold running, then the docker desktop app, then ran the docker command, then opened Open WebUI via my browser at localhost:3000, and then went into settings and added a "0" to the API key field.  Then went to test the chat functionality, and still got the error message about models.

So any ideas?  Thanks again.

**Reply by hectibrown** [reply]
> Does KoboldCPP_ROCM have an API Key? If so, where do I put it so I can then match it in WebUI? I've just looked through each setting tab in 1.67 Kobold_ROCM and I'm not seeing API key anywhere. Or do you mean I can put anything I want in there, and the system won't care cause there is no API Key in KoboldCPP? Thanks for any assistance you can offer! Cheers,
> 
> Update: Just got kobold running, then the docker desktop app, then ran the docker command, then opened Open WebUI via my browser at localhost:3000, and then went into settings and added a "0" to the API key field. Then went to test the chat functionality, and still got the error message about models.
> 
> So any ideas? Thanks again.

Hi, I found the Kobold API documentation and found that the following works:
![image](https://github.com/open-webui/open-webui/assets/51929979/8dd441c5-4700-482a-915d-0ab7fe346fa4)


the entire API can be seen here:
https://lite.koboldai.net/koboldcpp_api#/

**Reply by tukangcode** [reply]
Thanks this one  I need

**Comment by kaattaalan** [comment]
This guy compile ollama for amd GPUs with rocm support and has excellent guides on how to install
https://github.com/likelovewant/ollama-for-amd
